--- wabt/CMakeLists.txt.orig	2020-03-10 11:29:54.825459486 +0800
+++ wabt/CMakeLists.txt	2020-03-10 11:31:14.530946589 +0800
@@ -22,6 +22,7 @@
 option(BUILD_TESTS "Build GTest-based tests" ON)
 option(USE_SYSTEM_GTEST "Use system GTest, instead of building" OFF)
 option(BUILD_TOOLS "Build wabt commandline tools" ON)
+option(BUILD_LIBWASM "Build wasm C-API libwasm" OFF)
 option(USE_ASAN "Use address sanitizer" OFF)
 option(USE_MSAN "Use memory sanitizer" OFF)
 option(USE_LSAN "Use leak sanitizer" OFF)
@@ -319,6 +320,7 @@
 add_library(wabt STATIC ${WABT_LIBRARY_SRC})
 
 # libwasm, which implenents the wasm C API
+if(BUILD_LIBWASM)
 add_library(wasm SHARED ${WABT_LIBRARY_SRC} src/interp/interp-wasm-c-api.cc)
 target_link_libraries(wasm wabt)
 target_include_directories(wasm PUBLIC third_party/wasm-c-api/include)
@@ -329,6 +331,7 @@
   target_compile_definitions(wasm PRIVATE "WASM_API_EXTERN=__attribute__((visibility(\"default\")))")
 endif ()
 set_target_properties(wasm PROPERTIES CXX_VISIBILITY_PRESET hidden)
+endif()
 
 if (NOT EMSCRIPTEN)
   if (CODE_COVERAGE)
